components/infobox/wikis/stormgate/infobox_league_custom.lua

```diff
@@ -0,0 +1,228 @@
+
+	return league:createInfobox()
+end
+
+---@param args table
+---@return string?
+function CustomLeague._getStatus(args)
+	local status = args.status or Variables.varDefault('tournament_status')
+	if Logic.isNotEmpty(status) then
+		---@cast status -nil
+		return status:lower()
+	end
+
+	if Logic.readBool(_args.cancelled) then
+		return CANCELLED
+	end
+
+	if CustomLeague._isFinished(args) then
+		return FINISHED
+	end
+end
+
+---@param args table
+---@return boolean
+function CustomLeague._isFinished(args)
+	local finished = Logic.readBoolOrNil(args.finished)
+	if finished ~= nil then
+		return finished
+	end
+
+	local queryDate = _league:_cleanDate(args.edate) or _league:_cleanDate(args.date) or FALLBACK_DATE
+
+	if os.date('%Y-%m-%d') < queryDate then
+		return false
+	end
+
+	return mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = '[[pagename::' .. string.gsub(mw.title.getCurrentTitle().text, ' ', '_') .. ']] '
+			.. 'AND [[opponentname::!TBD]] AND [[placement::1]]',
+		query = 'date',
+		order = 'date asc',
+		limit = 1
+	})[1] ~= nil
+end
+
+---@return WidgetInjector
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		table.insert(widgets, Cell{name = 'Game Version', content = {CustomLeague._getGameVersion(_args)}})
+	elseif id == 'customcontent' then
+		if _args.player_number and _args.player_number > 0 then
+			Array.appendWith(widgets,
+				Title{name = 'Player Breakdown'},
+				Cell{name = 'Number of Players', content = {_args.raceBreakDown.total}},
+				Breakdown{content = _args.raceBreakDown.display, classes = { 'infobox-center' }}
+			)
+		end
+
+		--teams section
+		if Logic.isNumeric(_args.team_number) and tonumber(_args.team_number) > 0 then
+			Array.appendWith(widgets,
+				Title{name = 'Teams'},
+				Cell{name = 'Number of Teams', content = {_args.team_number}}
+			)
+		end
+
+		--maps
+		if String.isNotEmpty(_args.map1) then
+			Array.appendWith(widgets,
+				Title{name = 'Maps'},
+				Center{content = {CustomLeague._mapsDisplay(_args.maps)}}
+			)
+		end
+	end
+
+	return widgets
+end
+
+---@param maps {link: string, displayname: string}[]?
+---@return string?
+function CustomLeague._mapsDisplay(maps)
+	if not maps then return end
+
+	return table.concat(
+		Array.map(maps, function(mapData)
+			return tostring(CustomLeague:_createNoWrappingSpan(
+				PageLink.makeInternalLink({}, mapData.displayname, mapData.link)
+			))
+		end),
+		'&nbsp;• '
+	)
+end
+
+---@param args table
+---@return string?
+function CustomLeague._getGameVersion(args)
+	if not args.patch then
+		return
+	end
+
+	local gameVersion = '[[' .. args.patch .. ']]'
+	if not args.epatch or args.epatch == args.patch then
+		return gameVersion
+	end
+
+	return gameVersion .. ' &ndash; [[' .. args.epatch .. ']]'
```

Comment:
- errors for set epatch with unset patch
- if both are unset returns empty sting forcing the cell to show which is not intended imo
- if patch = epatch it would still display `x.yy - x.yy` instead of just `x.´yy`

Reply:
First and Third are fine I think
Second is a real issue, I agree

Reply:
Are you sure about the second one?
```
local Page = require('Module:Page')
mw.logObject(table.concat({Page.makeInternalLink(patch), Page.makeInternalLink(epatch)}, ' &ndash; ')) -- ""
``` 
returns empty string, which the build of Cell ignores? https://github.com/Liquipedia/Lua-Modules/blob/59a2edc62477949e967ec7840f501020ffa778bc/components/infobox/commons/infobox_widget_cell.lua#L65

Reply:
ah we do check against empty string, then nvm the second

Reply:
```suggestion
	return table.concat({
		Page.makeInternalLink(args.patch),
		Page.makeInternalLink(args.epatch ~= args.patch and args.epatch or nil)
	}, ' &ndash; ')
```

Reply:
and skip `and args.patch` right?

Reply:
that is just the nil check so it can not error